File: Wifi Range [19 - April - 2023]/Approach 3(Optimal).cpp
// >> C++ CODE <<


bool wifiRange(int N, string S, int X){
        int st = 0;
        while (st < N && S[st] != '1') {
            st += 1;
        }
        if (st == N || st > X) {
            return false;
        }
        int last = st;
        for (int i = st; i < N; i++) {
            if (i - last + 1 > 2 * (X + 1)) {
                //cout << last << " and " << i << endl;
                return false;
            }
            if (S[i] == '1') {
                last = i;
            }
        }
        if (N - last > X + 1) {
            return false;
        }
        return true;
    }

File: Wifi Range [19 - April - 2023]/Approach 3(Optimal).py
        #   >> PYTHON CODE <<
	  
        st=0
        while st<N and S[st]!='1':
            st+=1


        if(st==N or st>X):
            return False

        last=st
        for i in range(st,N):
            if((i-last+1)>2*(X+1)):
                #print("{} and {}".format(last,i))
                return False
            if(S[i]=='1'):
                last=i
        
        if(N-last>(X+1)):
            return False
            
        return True